pkg/sql/schemachanger/scplan/internal/rules/current/helpers.go

```diff
@@ -207,6 +207,14 @@ func isConstraint(e scpb.Element) bool {
 	return false
 }
 
+func isCrossDescriptorConstraint(e scpb.Element) bool {
```

Comment:
use some words to explain this. Our taxonomy is arbitrary and could benefit from any sort of extra words we can sprinkle onto it.

Reply:
done

Reply:
Also, these filters should have tests in `assertions_test.go`. Please don't skimp on these. They help keep this logic tight, reliable and future-proof.

Reply:
I added a test there to assert that we have multiple partitions for constraints: is it cross-descriptor or not? Is it index-backed or not? etc.